pkg/services/gc/gc.go

```diff
@@ -100,48 +101,97 @@ func (gc *GC) Start(ctx context.Context) error {
 	return nil
 }
 
+type runOptions struct {
+	typePredicate   func(context.Context, schema.GroupVersionKind) (bool, error)
+	objectPredicate func(context.Context, unstructured.Unstructured) (bool, error)
+}
+
+type RunOptionsFn func(*runOptions)
+
+func WithObjectFilter(fn func(context.Context, unstructured.Unstructured) (bool, error)) RunOptionsFn {
+	return func(o *runOptions) {
+		o.objectPredicate = fn
+	}
+}
+func WithTypeFilter(fn func(context.Context, schema.GroupVersionKind) (bool, error)) RunOptionsFn {
+	return func(o *runOptions) {
+		o.typePredicate = fn
+	}
+}
+
+func (gc *GC) listResources(
+	ctx context.Context,
+	res Resource,
+	opts metav1.ListOptions,
+) ([]unstructured.Unstructured, error) {
+	items, err := gc.client.Dynamic().Resource(res.GroupVersionResource()).Namespace("").List(ctx, opts)
+	switch {
+	case k8serr.IsForbidden(err) || k8serr.IsMethodNotSupported(err):
+		gc.log(ctx).V(3).Info(
+			"cannot list resource",
+			"reason", err.Error(),
+			"gvk", res.GroupVersionKind(),
+		)
+
+		return nil, nil
+	case err != nil:
+		return nil, err
+	default:
+		return items.Items, nil
```

Comment:
I remember something complained that there is no return if I do it just in default. But looks like not the case anymore